config/web.go

```diff
@@ -26,9 +31,38 @@ func (web *webConfig) validateAndSetDefaults() {
 	} else if web.Port < 0 || web.Port > math.MaxUint16 {
 		panic(fmt.Sprintf("port has an invalid: value should be between %d and %d", 0, math.MaxUint16))
 	}
+
+	web.ContextRoot = validateAndBuild(web.ContextRoot)
+}
+
+// validateAndBuild validates and builds a checked
+// path for the context root
+func validateAndBuild(contextRoot string) string {
+	trimedContextRoot := strings.Trim(contextRoot, "/")
+
+	if len(trimedContextRoot) == 0 {
+		return DefaultContextRoot
+	} else {
+		url, err := url.Parse(trimedContextRoot)
+		if err != nil {
+			panic(fmt.Sprintf("Invalid context root %s - error: %s.", contextRoot, err))
+		}
+		if url.Path != trimedContextRoot {
+			panic(fmt.Sprintf("Invalid context root %s, simple path required.", contextRoot))
+		}
+
+		return "/" + strings.Trim(url.Path, "/") + "/"
+	}
 }
 
 // SocketAddress returns the combination of the Address and the Port
 func (web *webConfig) SocketAddress() string {
 	return fmt.Sprintf("%s:%d", web.Address, web.Port)
 }
+
+// AppendToContexRoot appends the given string to the context root
+// AppendToContexRoot takes care of having only one "/" character at
+// the join point and exactly on "/" at the end
+func (web *webConfig) AppendToContexRoot(fragment string) string {
```

Comment:
Seems like you've made a small typo in the function name (`ContexRoot` instead of `ContextRoot`)

That said, I think renaming this function to `PrependWithContextRoot` would be a bit more descriptive, if you don't mind

Reply:
Thanks for finding the typo - I renamed the method as requested.